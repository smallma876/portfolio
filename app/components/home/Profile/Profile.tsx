import React from "react";
import GiantText from "../../../../components/GiantText/GiantText";
import TypingEffect from "../../../../components/TypingEffect/TypingEffect";
import { Palette } from "../../../../utilities/palette";

const Profile = () => {
  return (
    <>
      <TypingEffect component="h1" variant="h1" text="I'm Sergio" />
      <GiantText color={Palette.DarkYellow}>Front End Developer</GiantText>
    </>
  );
};

export default Profile;
