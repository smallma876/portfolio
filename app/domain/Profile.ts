export interface Profile {
  title: string;
  imageProfile: {
    fields: {
      file: {
        url: string;
      };
    };
  };
}

