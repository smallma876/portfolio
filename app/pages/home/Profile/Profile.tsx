import Image from "next/image";
import React, { FC } from "react";
import GiantText from "../../../../components/GiantText/GiantText";
import TypingEffect from "../../../../components/TypingEffect/TypingEffect";
import { Box, Flexbox } from "../../../../shared/components/Flexbox/Flexbox";
import { Palette } from "../../../../utils/palette";

const languagesProgramming = [
  {
    src: "./images/it/css3-icon.svg",
    alt: "css3",
  },
  {
    src: "./images/it/html-icon.svg",
    alt: "html",
  },
  {
    src: "./images/it/javascript-icon.svg",
    alt: "javascript",
  },
  {
    src: "./images/it/reactjs-icon.svg",
    alt: "reactjs",
  },
  {
    src: "./images/it/angular-icon.svg",
    alt: "angular",
  },
  {
    src: "./images/it/typescript-icon.svg",
    alt: "typescript",
  },
  {
    src: "./images/it/nextjs-icon.svg",
    alt: "nextjs",
  },
  {
    src: "./images/it/git-icon.svg",
    alt: "git",
  },
];

interface ProfileProps {
  title: string;
}

const Profile: FC<ProfileProps> = ({ title }) => {
  return (
    <>
      <TypingEffect component="h1" variant="h1" text="I'm Sergio" />
      <GiantText color={Palette.White}>{title}</GiantText>
      <Flexbox gap="0.5rem">
        {languagesProgramming.map(({ src, alt }) => (
          <Box key={alt}>
            <Image src={src} alt={alt} width={30} height={30} />
          </Box>
        ))}
      </Flexbox>
    </>
  );
};

export default Profile;
