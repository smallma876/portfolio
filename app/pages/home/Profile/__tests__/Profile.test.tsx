import { render, screen } from "@testing-library/react";
import Profile from "../Profile";

const renderComponent = () => {
  const { container } = render(<Profile title="hola mundo" />);

  return container;
};

describe("Profile", () => {
  it("should render match snapshot", () => {
    const container = renderComponent();
    expect(screen.getByText("hola mundo")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
