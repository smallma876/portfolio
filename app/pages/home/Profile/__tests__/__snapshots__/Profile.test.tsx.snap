// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile should render match snapshot 1`] = `
<div>
  <div
    class="sc-gswNZR bybFwM"
  >
    <h1 />
  </div>
  <h1
    class="sc-bcXHqe sVGpq"
    color="#FFFFFF"
  >
    hola mundo
  </h1>
  <div
    class="sc-dkrFOg lSzhK"
  >
    <div
      class="sc-hLBbgP"
    >
      <img
        alt="css3"
        data-nimg="1"
        decoding="async"
        height="30"
        loading="lazy"
        src="./images/it/css3-icon.svg"
        style="color: transparent;"
        width="30"
      />
    </div>
    <div
      class="sc-hLBbgP"
    >
      <img
        alt="html"
        data-nimg="1"
        decoding="async"
        height="30"
        loading="lazy"
        src="./images/it/html-icon.svg"
        style="color: transparent;"
        width="30"
      />
    </div>
    <div
      class="sc-hLBbgP"
    >
      <img
        alt="javascript"
        data-nimg="1"
        decoding="async"
        height="30"
        loading="lazy"
        src="./images/it/javascript-icon.svg"
        style="color: transparent;"
        width="30"
      />
    </div>
    <div
      class="sc-hLBbgP"
    >
      <img
        alt="reactjs"
        data-nimg="1"
        decoding="async"
        height="30"
        loading="lazy"
        src="./images/it/reactjs-icon.svg"
        style="color: transparent;"
        width="30"
      />
    </div>
    <div
      class="sc-hLBbgP"
    >
      <img
        alt="angular"
        data-nimg="1"
        decoding="async"
        height="30"
        loading="lazy"
        src="./images/it/angular-icon.svg"
        style="color: transparent;"
        width="30"
      />
    </div>
    <div
      class="sc-hLBbgP"
    >
      <img
        alt="typescript"
        data-nimg="1"
        decoding="async"
        height="30"
        loading="lazy"
        src="./images/it/typescript-icon.svg"
        style="color: transparent;"
        width="30"
      />
    </div>
    <div
      class="sc-hLBbgP"
    >
      <img
        alt="nextjs"
        data-nimg="1"
        decoding="async"
        height="30"
        loading="lazy"
        src="./images/it/nextjs-icon.svg"
        style="color: transparent;"
        width="30"
      />
    </div>
    <div
      class="sc-hLBbgP"
    >
      <img
        alt="git"
        data-nimg="1"
        decoding="async"
        height="30"
        loading="lazy"
        src="./images/it/git-icon.svg"
        style="color: transparent;"
        width="30"
      />
    </div>
  </div>
</div>
`;
